src/shared/form/util.ts

```diff
@@ -1,10 +1,14 @@
-import { FormContextValues } from "react-hook-form";
-import { APIError } from "../types";
+import type { FormContextValues } from "react-hook-form";
+import type { APIError } from "../types";
 
-export const cleanFormData = <T>(data: T): Partial<T> => {
-  return Object.fromEntries(
-    Object.entries(data).filter(([key, value]) => value !== "")
-  );
+export const cleanFormData = <T extends {}>(data: T): Partial<T> => {
```

Comment:
Can you clarify what the differences are between this function and the old one?

Reply:
Functionally, there's no difference. The reason for the change is that Typescript changed the typing of `Object.entries` to return the equivalent of `{ [k: any]: any }` instead of `{ [k keyof T]: T[k] }`. Now the only way to iterate over an object and ensure keys are typed `keyof T` is with a for-in loop, hence the change.